src/java/org/apache/ivy/core/cache/DefaultRepositoryCacheManager.java

```diff
@@ -1215,26 +1217,22 @@ public ResolvedResource resolve(Artifact artifact) {
                     for (int i = 0; i < confs.length; i++) {
                         Artifact[] arts = md.getArtifacts(confs[i]);
                         for (int j = 0; j < arts.length; j++) {
-                            Artifact transformedArtifact = NameSpaceHelper.transform(
-                                arts[j], options.getNamespace().getToSystemTransformer());
-                            ArtifactOrigin origin = getSavedArtifactOrigin(
-                                transformedArtifact);
-                            File artFile = getArchiveFileInCache(
-                                transformedArtifact, origin, false);
-                            if (artFile.exists()) {
-                                Message.debug("deleting " + artFile);
-                                if (!artFile.delete()) {
-                                    // Old artifacts couldn't get deleted!
-                                    // Restore the original ivy file so the next time we
-                                    // resolve the old artifacts are deleted again
-                                    backupDownloader.restore();
-                                    Message.error("Couldn't delete outdated artifact from cache: " + artFile);
-                                    return null;
-                                }
-                            }
-                            removeSavedArtifactOrigin(transformedArtifact);
+                            if (!prepAndDeleteArtifact(arts[j], options, backupDownloader)) return null;
                         }
                     }
+
+                    final Artifact sourceArtifact = new MDArtifact(md, mrid.getName(), "src", "jar", null,
+                        Collections.singletonMap("m:classifier", "sources"));
+
+                    final Artifact srcArtifact = new MDArtifact(md, mrid.getName(), "src", "jar", null,
+                        Collections.singletonMap("m:classifier", "src"));
+
+                    final Artifact javadocArtifact = new MDArtifact(md, mrid.getName(), "doc", "jar", null,
+                        Collections.singletonMap("m:classifier", "javadoc"));
```

Comment:
a really good question.   I think the differences represent something we'll want to clean-up/unify.   It may be a good idea to move that correspondence map onto this side and then use it in sbt, so we can correct those issues as needed.